src/api/docs/content/specs/dns.yaml

```diff
@@ -0,0 +1,169 @@
+openapi: 3.0.2
+components:
+  paths:
+    blocking:
+      summary: Modify blocking
+      get:
+        summary: Get current blocking status
+        tags:
+          - DNS control
+        operationId: "get_blocking"
+        security:
+          - sidHeader: []
+        description: |
+          The property `timer` may contain additional details concerning a temporary en-/disabling.
+          It is `null` when no timer is active (the current status is permanent).
+        responses:
+          '200':
+            description: OK
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/blocking'
+                    - $ref: '#/components/schemas/timer'
+      post:
+        summary: Change current blocking status
+        tags:
+          - DNS control
+        operationId: "set_blocking"
+        description: |
+          Change the current blocking mode by setting `blocking` to the desired value.
+          The optional `timer` object may used to set a timer. Once this timer elapsed, the opposite blocking mode is automatically set.
+          For instance, you can request `{blocking: true, timer: 60}` to disable Pi-hole for one minute.
+          Blocking will be automatically resumed afterwards.
+
+          You can terminate a possibly running timer by setting `timer` to `null` (the set mode becomes permanent).
+        requestBody:
+          description: Callback payload
+          content:
+            'application/json':
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/blocking'
+                  - $ref: '#/components/schemas/timer'
+        responses:
+          '200':
+            description: OK
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/blocking'
+                    - $ref: '#/components/schemas/timer'
+          '400':
+            description: Bad request
+            content:
+              application/json:
+                schema:
+                  oneOf:
+                  - $ref: '#/components/schemas/errors/no_payload'
+                  - $ref: '#/components/schemas/errors/item_missing'
+          '401':
+            description: Unauthorized
+            content:
+              application/json:
+                schema:
+                  $ref: 'common.yaml#/components/errors/unauthorized'
+    cache:
```

Comment:
`/dns/cache` 
retrieves more than `.cache_size`, `.cache_inserted` and `cache_evited`